tests/testsuite/build.rs

```diff
@@ -1563,6 +1579,372 @@ fn crate_env_vars() {
+            "src/lib.rs",
+            r#"
+                use std::env;
+                #[test]
+                fn env() {
+                    // Check that CARGO_WORKSPACE_DIR isn't set for unit tests
+                    assert!(option_env!("CARGO_WORKSPACE_DIR").is_none());
+                    env::var("CARGO_WORKSPACE_DIR").unwrap_err();
+                }
+            "#,
+        )
+        .file(
+            "examples/ex-env-vars.rs",
+            r#"
+                fn main() {
+                    // Verify CARGO_WORKSPACE_DIR isn't set for examples
+                    assert!(option_env!("CARGO_WORKSPACE_DIR").is_none());
+                }
+            "#,
+        )
+        .file(
+            "tests/env.rs",
+            r#"
+                use std::path::Path;
+
+                #[test]
+                fn env() {
+                    assert!(Path::new(option_env!("CARGO_WORKSPACE_DIR").unwrap()).join(file!()).exists());
+                }
+            "#,
+        )
+        // path dependency
+        .file(
+            "baz/Cargo.toml",
+            r#"
+            [workspace]
+            members = ["baz_member"]
+            "#,
+        )
+        .file(
+            "baz/baz_member/Cargo.toml",
+            r#"
+            [package]
+            name = "baz_member"
+            version = "0.1.0"
+            authors = []
+            "#,
+        )
+        .file("baz/baz_member/src/lib.rs", "#[test] fn baz_member() {}")
+        .file(
+            "baz/baz_member/tests/env.rs",
+            r#"
+            use std::path::Path;
+
+            #[test]
+            fn env() {
+                // is_primary = true
+                // is_workspace = false
+                // cargo_workspace_dir = "."
+                // cx.bcx.ws.root() = "/Users/yerke/dev/rust/cargo/target/tmp/cit/t0/foo"
+                // unit.pkg.root() = "/Users/yerke/dev/rust/cargo/target/tmp/cit/t0/foo/baz/baz_member"
+                // option_env!("CARGO_WORKSPACE_DIR").unwrap() = "."
+                // file!() = "baz/baz_member/tests/env.rs"
+                // std::env::current_dir().unwrap() = "/Users/yerke/dev/rust/cargo/target/tmp/cit/t0/foo/baz/baz_member"
+                // option_env!("CARGO_MANIFEST_DIR").unwrap() = "/Users/yerke/dev/rust/cargo/target/tmp/cit/t0/foo/baz/baz_member"
+                // thread 'env' panicked at 'assertion failed: Path::new(option_env!(\"CARGO_WORKSPACE_DIR\").unwrap()).join(file!()).exists()', baz/baz_member/tests/env.rs:15:17
+                // since it's trying to access "/Users/yerke/dev/rust/cargo/target/tmp/cit/t0/foo/baz/baz_member/baz/baz_member/tests/env.rs"
+                dbg!(option_env!("CARGO_WORKSPACE_DIR").unwrap());
+                dbg!(file!());
+                dbg!(std::env::current_dir().unwrap());
+                dbg!(option_env!("CARGO_MANIFEST_DIR").unwrap());
+                assert!(Path::new(option_env!("CARGO_WORKSPACE_DIR").unwrap()).join(file!()).exists());
```

Comment:
This test fails. I added debug info above. In short, we are not recognizing that path dependency is a part of its own workspace.